### ISSUE TITLE
cannot instantiate SegmentationKS for training

### ISSUE THREAD

**Issue by kashefy** [TWOEARS/examples#3]
Hey @chrschy, I was trying out demo for training SegmentationKS and ran into an error. The constructor checks if the model file exists and produces an error if it cannot find it. This happens [here](https://github.com/TWOEARS/blackboard-system/blob/master/src/knowledge_sources/SegmentationKS.m#L164).

In order to train SegmentationKS I would still need to create an instance and it will not get past this line. Or am I missing something?

If this is indeed the case, I was thinking we could
1. pass a flag to the constructor to indicate that we're instantiating the object in "train mode" and skip this check accordingly.
2. Turn the train method into a static method or a factory method that does not need the object.

Option 1 seems simpler. What do you think?

**Comment by hagenw** [comment]
Hi @kashefy,

what exactly did you run to produce your error?

When I go to the [segmentation folder of TWOEARS/examples](https://github.com/TWOEARS/examples/tree/master/segmentation) and run `demo_train_segmentation` everything runs fine for me.

**Comment by kashefy** [comment]
Hi @hagenw,
I'm working off of `demo_train_segmentation` as well. I get the error when changing one of the KS's [parameters](https://github.com/TWOEARS/examples/blob/master/segmentation/demo_train_segmentation.m#L21).

A change of parameters involves re-training the KS. Changing the parameters changes the object's reqHashs attribute. The hash value determines the model filename. If you run the train script without changing the parameters you are basically re-training an existing model. In this case the constructor finds the existing model and no errors occur. I changed the `nChannels` parameter from 16 to 32. I think this leads to a different hash value for which a model does not exist yet.

I wanted to change `nChannels` to better align with the features parameters required for the IdentityKS downstream.

**Comment by hagenw** [comment]
Yes, I can recreate this. Change `nChannels` after running the demo (in my case from 32 to 16) results in:

``` Matlab
>> demo_train_segmentation
Downloading file https://dev.qu.tu-berlin.de/projects/twoears-database/repository/revisions/master/raw//learned_models/SegmentationKS/DemoKS/DemoKS_models_66cc1d312cb06e04624ffd91283691a0d6d3ec028e79306c718aee17dd2b922b.mat
Error using SegmentationKS (line 168)
No trained models are available for this KS. Please ensure to run KS training first.

Error in demo_train_segmentation (line 33)
segKS = SegmentationKS(ksName, ...

```
